### ISSUE TITLE
Error: aerender.exe failed to render the output into the file due to an unknown reason

### ISSUE THREAD

**Issue by jjdenhertog** [inlife/nexrender#1016]
**Describe the bug**
Somewhere down the line nexrender stopped working and it seems to be impossible to restore it to a working state anymore. I tried installing it through NPM and tried using the binaries. I also tried reinstalling AE en AME. Resetting both apps and Creative Cloud.

This command:
```bash
./nexrender-cli-macos "{\"template\":{\"src\":\"file:/$PWD/project.aep\",\"composition\":\"1x1\", \"outputModule\": \"molo loco video\", \"outputExt\":\"mp4\"}}"
```

Gives the following output:
```bash
> starting nexrender-cli
using automatically determined directory of After Effects installation:
 - /Applications/Adobe After Effects 2024/aerender
checking After Effects command line renderer patch...
command line patch already is in place
[9GA8ksi4rln7kjrfZnK8f] setting up job...
[9GA8ksi4rln7kjrfZnK8f] working directory is: /var/folders/k0/8trjf08x68jcqpc51p3970m00000gn/T/nexrender/9GA8ksi4rln7kjrfZnK8f
[9GA8ksi4rln7kjrfZnK8f] applying predownload actions...
[9GA8ksi4rln7kjrfZnK8f] downloading assets...
[9GA8ksi4rln7kjrfZnK8f] applying postdownload actions...
[9GA8ksi4rln7kjrfZnK8f] applying prerender actions...
[9GA8ksi4rln7kjrfZnK8f] running script assemble...
[9GA8ksi4rln7kjrfZnK8f] rendering job...
[9GA8ksi4rln7kjrfZnK8f] setting aerender log path to project folder: /var/folders/k0/8trjf08x68jcqpc51p3970m00000gn/T/nexrender/9GA8ksi4rln7kjrfZnK8f/aerender.log
catching the error internally
[9GA8ksi4rln7kjrfZnK8f] cleaning up...
> job rendering failed
Error: aerender.exe failed to render the output into the file due to an unknown reason
    at ChildProcess.<anonymous> (/snapshot/nexrender/packages/nexrender-core/src/tasks/render.js)
    at ChildProcess.emit (node:events:537:28)
    at maybeClose (node:internal/child_process:1091:16)
    at Socket.<anonymous> (node:internal/child_process:449:11)
    at Socket.emit (node:events:537:28)
    at Pipe.<anonymous> (node:net:747:14)
```

But when I run this command everything works:
```bash
 /Applications/Adobe\ After\ Effects\ 2024/aerender -project "$PWD/project.aep" -comp "1x1"  -OMtemplate "molo loco video" -output "$PWD/result.mp4"
```

I get this output:
```
aerender version 24.5x52
PROGRESS: Launching After Effects...
PROGRESS: Adding specified comp to Render Queue
PROGRESS:  12/08/2024 12:39:16 CEST: Starting composition “1x1”.

PROGRESS:  Render Settings: Best Settings
PROGRESS:  Quality: Best
PROGRESS:  Resolution: Full
PROGRESS:  Size: 1920 x 1080
PROGRESS:  Proxy Use: Use No Proxies
PROGRESS:  Effects: Current Settings
PROGRESS:  Disk Cache: Read Only
PROGRESS:  Color Depth: Current Settings
PROGRESS:  Frame Blending: On for Checked Layers
PROGRESS:  Field Render: Off
PROGRESS:  Pulldown: Off
PROGRESS:  Motion Blur: On for Checked Layers
PROGRESS:  Solos: Current Settings
PROGRESS:  Time Span: Work Area
PROGRESS:  Start: 0;00;00;00
PROGRESS:  End: 0;00;29;29
PROGRESS:  Duration: 0;00;30;00
PROGRESS:  Frame Rate: 29,97 (comp)
PROGRESS:  Guide Layers: All Off
PROGRESS:  Skip Existing Files: Off
PROGRESS:
PROGRESS:  Output Module: molo loco video
PROGRESS:  Output To: /Users/jjdenhertog/Documents/Projects/molo-render/patch/result.mp4
PROGRESS:  Format: H.264
PROGRESS:  Output Info: VBR, Target 15,00 Mbps
PROGRESS:  Include: Project Link
PROGRESS:  Output Audio: On (if comp has audio)
PROGRESS:  Channels: RGB
PROGRESS:  Depth: Millions of Colors
PROGRESS:  Color: Premultiplied
PROGRESS:  Resize: -
PROGRESS:  Crop: -
PROGRESS:  Final Size: 1920 x 1080
PROGRESS:  Profile: -
PROGRESS:  Post-Render Action: None
PROGRESS:
PROGRESS:  0;00;00;00 (1): 0 Seconds
```

**Information about environment**
* Which version of nexrender are you using?
1.53.1

* What kind of the setup do you have? (server+worker or cli)
CLI

* Are you using pre-built binaries or globally installed npm packages?
Tested with both

* Which kind (and version) of operating system you are running it in?
Apple M1 Max
14.1.1 (23B81)

* Any additional information that might be helpful
There are no logs created from AE. So it feels like the aerender process is never executed.

**Comment by jjdenhertog** [comment]
I have been debugging this line in render.js

```JavaScript
instance.on('close', (code) => 
```

And found by adding the signal that it's crashing on a SIGILL signal. And found with some more research that it seems to be opening the aerender binary using Rosetta. And that gives the error "illegal instructions: 4". 

I can reproduce it by opening Terminal in Rosetta and running this command which was previously working:
```bash
 /Applications/Adobe\ After\ Effects\ 2024/aerender -project "$PWD/project.aep" -comp "1x1"  -OMtemplate "molo loco video" -output "$PWD/result.mp4"
```

Now (in Rosetta) it gives this output:
```bash
Illegal instruction: 4
```

So my question is, how can I make sure that the command that is spawned in render.js is not using Rosetta?

**Comment by jjdenhertog** [comment]
Alright, I managed to solve the issue myself. So for anyone running into the same issue.

It was caused by Node running under Rosette. I originally installed the node versions while running Terminal in Rosetta. As a result the Node version was also running in Rosetta. 

This command returned `x64` rather than `arm64`.
```bash
node -e 'console.log(process.arch)'
```

I reinstalled the node version while running the terminal in Rosetta and everything worked like a charm.
